src/main/java/com/google/api/codegen/util/php/PhpPackageUtil.java

```diff
@@ -62,4 +50,34 @@ public static String getFullyQualifiedName(String packageName, String objectName
   public static boolean isPackageVersion(String versionString) {
     return versionString.matches(PACKAGE_VERSION_REGEX);
   }
+
+  public static String getPackageNameFromVersionOnwards(String packageName) {
+    ArrayList<String> packageComponents = new ArrayList<>();
+    List<String> pieces = Arrays.asList(PhpPackageUtil.splitPackageName(packageName));
+    boolean foundVersion = false;
+    for (String packageElement : Lists.reverse(pieces)) {
+      packageComponents.add(packageElement);
+      if (isPackageVersion(packageElement)) {
+        foundVersion = true;
+        break;
+      }
+    }
+    if (foundVersion) {
+      return buildPackageName(Lists.reverse(packageComponents));
+    } else {
+      // If we did not find a version, then "FromVersionOnwards" is null
+      return null;
+    }
+  }
+
+  public static String getPackageNameBeforeVersion(String packageName) {
```

Comment:
It feels like the more fundamental thing you want is the "base package name", which everything else is a subpackage of. I think it would feel more intuitive to name things in terms of that.

Reply:
Updated to use "base package name" language, PTAL